src/server/services/crypto.js

```diff
@@ -9,3 +11,33 @@ export async function hashPassword(password) {
 export async function comparePasswords(password, hash) {
   return await bcrypt.compare(password, hash);
 }
+
+export function hashSecret(str) {
```

Comment:
Strings used here will still need a "secret" arg, as in 
```
web3.eth.accounts.sign(
    str + randomArg,
    process.env.HASH_PRIVATE_KEY,
  );
```
otherwise they can be brute forced with the public key from known titles. 

I'd suggest adding a third arg here for the random part and a `if (!randomArg) throw` to make sure it doesn't get forgotten

Reply:
worth noting maybe that we could just go back to a simple hash too, like `web3.eth.sha3(str + randomArg)`

Reply:
Then we would only use it to identify it with out entries in the database though and not  to make it reproducable or verifiable if I understand correctly?

Reply:
We have to publish `randomArg` for it to be verifiable/reproduceable - the idea is that Culturestake releases these after the event for only the top three artworks. This way if we use different randomArgs for all the ids, we can selectively choose which artworks are identifiable